scalafmt-dynamic/src/test/scala/tests/ScalafmtVersionSuite.scala

```diff
@@ -68,16 +103,38 @@ class ScalafmtVersionSuite extends AnyFunSuite {
   }
 
   test("order versions") {
-    assert(ScalafmtVersion(2, 0, 0, 0) > ScalafmtVersion(1, 5, 1, 0))
-    assert(ScalafmtVersion(2, 0, 0, 0) > ScalafmtVersion(2, 0, 0, 1))
-    assert(ScalafmtVersion(2, 0, 0, 4) > ScalafmtVersion(1, 9, 9, 9))
-    assert(ScalafmtVersion(0, 1, 2, 0) > ScalafmtVersion(0, 1, 1, 0))
-    assert(ScalafmtVersion(0, 2, 2, 0) > ScalafmtVersion(0, 1, 2, 0))
-    assert(ScalafmtVersion(2, 0, 0, 2) < ScalafmtVersion(2, 0, 0, 4))
-    assert(ScalafmtVersion(2, 0, 0, 2) < ScalafmtVersion(2, 0, 0, 0))
-    assert(ScalafmtVersion(0, 1, 2, 0) < ScalafmtVersion(1, 0, 0, 0))
-    assert(ScalafmtVersion(0, 1, 8, 0) < ScalafmtVersion(0, 2, 2, 0))
-    assert(!(ScalafmtVersion(2, 0, 0, 0) < ScalafmtVersion(1, 5, 1, 0)))
-    assert(!(ScalafmtVersion(0, 1, 8, 0) > ScalafmtVersion(0, 2, 2, 0)))
+    assert(
+      ScalafmtVersion(2, 0, 0, 0, false) > ScalafmtVersion(1, 5, 1, 0, false)
+    )
+    assert(
+      ScalafmtVersion(2, 0, 0, 0, false) > ScalafmtVersion(2, 0, 0, 1, false)
+    )
+    assert(
+      ScalafmtVersion(2, 0, 0, 4, false) > ScalafmtVersion(1, 9, 9, 9, false)
+    )
+    assert(
+      ScalafmtVersion(0, 1, 2, 0, false) > ScalafmtVersion(0, 1, 1, 0, false)
+    )
+    assert(
+      ScalafmtVersion(0, 2, 2, 0, false) > ScalafmtVersion(0, 1, 2, 0, false)
+    )
+    assert(
+      ScalafmtVersion(2, 0, 0, 2, false) < ScalafmtVersion(2, 0, 0, 4, false)
+    )
+    assert(
+      ScalafmtVersion(2, 0, 0, 2, false) < ScalafmtVersion(2, 0, 0, 0, false)
+    )
+    assert(
+      ScalafmtVersion(0, 1, 2, 0, false) < ScalafmtVersion(1, 0, 0, 0, false)
+    )
+    assert(
+      ScalafmtVersion(0, 1, 8, 0, false) < ScalafmtVersion(0, 2, 2, 0, false)
+    )
+    assert(
+      !(ScalafmtVersion(2, 0, 0, 0, false) < ScalafmtVersion(1, 5, 1, 0, false))
+    )
+    assert(
+      !(ScalafmtVersion(0, 1, 8, 0, false) > ScalafmtVersion(0, 2, 2, 0, false))
+    )
```

Comment:
According to semantic versioning spec, we might should define the order `2.2.2 > 2.2.2-SNAPSHOT > 2.2.1` or something.
However, considering the use-case of `ScalafmtVersion` in scalafmt-dynamic, we don't need to implement such versioning I guess :)

Reply:
May be cover this case? It's not looking hard:

```scala
def <(other: ScalafmtVersion): Boolean =
    if (integerRepr == other.integerRepr && rc == other.rc)
      snapshot && !other.snapshot
    else if (integerRepr == other.integerRepr)
      rc != 0 && (other.rc == 0 || rc < other.rc)	   
    else integerRepr < other.integerRepr	  
```

Reply:
see suggestion above. if you do implement it, adding a test to cover your proposed case would be ideal.